core/state_transition.go

```diff
@@ -439,15 +441,25 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	// - reset transient storage(eip 1153)
 	st.state.Prepare(rules, msg.From, st.evm.Context.Coinbase, msg.To, vm.ActivePrecompiles(rules), msg.AccessList)
 
+	if !contractCreation {
```

Comment:
Why did you move this here?

Reply:
I think this is an artifact from the whole special casing when the eof initcode is invalid. I will revert it